packages/back-end/src/controllers/stripe.ts

```diff
@@ -4,106 +4,141 @@ import {
-        : null,
-      status: subscription.status,
-    },
-  });
-}
-export async function postStartTrial(
-  req: AuthRequest<{ qty: number; name: string }>,
+export async function postNewSubscription(
+  req: AuthRequest<{ qty: number; restart: boolean }>,
   res: Response
 ) {
-  const { qty, name } = req.body;
-
-  // If user already has a subscription, return immediately
-  if (req.organization?.subscription?.id) {
-    return res.status(200).json({
-      status: 200,
-    });
-  }
+  const { qty, restart } = req.body;
 
   try {
-    // Create organization first if needed
-    if (!req.organization) {
-      if (name.length < 3) {
-        throw new Error("Company name must be at least 3 characters long");
+    req.checkPermissions("organizationSettings");
+
+    const { org } = getOrgFromReq(req);
+
+    if (!org) {
+      throw new Error("No organization found");
+    }
+
+    const qtyInDb = org.members.length + (org.invites?.length || 0);
+
+    if (restart) {
+      if (qty !== qtyInDb) {
+        throw new Error("Error with quantity of users");
+      }
+    } else {
+      if (qty !== qtyInDb + 1) {
+        throw new Error("Error with quantity of users");
       }
-      req.organization = await createOrganization(
-        req.email || "",
-        req.userId || "",
-        name,
-        ""
-      );
     }
 
-    // Create customer in Stripe if not exists
-    if (!req.organization.stripeCustomerId) {
-      const resp = await stripe.customers.create({
-        email: req.email || "",
-        name: req.name || "",
+    let stripeCustomerId: string;
+
+    if (org.stripeCustomerId) {
+      stripeCustomerId = org.stripeCustomerId;
+    } else {
+      const { id } = await stripe.customers.create({
         metadata: {
-          user: req.userId || "",
-          organization: req.organization.id,
+          growthBookId: org.id,
         },
       });
-      req.organization.stripeCustomerId = resp.id;
-
-      await updateOrganization(req.organization.id, {
-        stripeCustomerId: resp.id,
-      });
+      stripeCustomerId = id;
     }
 
-    // Start subscription trial without payment method
-    const subscription = await stripe.subscriptions.create({
-      customer: req.organization.stripeCustomerId,
-      coupon: STRIPE_DEFAULT_COUPON,
-      collection_method: "charge_automatically",
-      trial_from_plan: true,
-      metadata: {
-        user: req.userId || "",
-        organization: req.organization.id,
-      },
-      items: [
+    await updateOrganization(org.id, {
+      stripeCustomerId: stripeCustomerId,
+    });
+
+    const session = await stripe.checkout.sessions.create({
+      mode: "subscription",
+      payment_method_types: ["card"],
+      customer: stripeCustomerId,
+      line_items: [
         {
-          price: STRIPE_PRICE,
+          price: org?.priceId || STRIPE_PRICE,
           quantity: qty,
         },
       ],
+      allow_promotion_codes: true,
```

Comment:
Does this show a "coupon code" field in the checkout?  If so, we might want to disable that for now.

Reply:
It does, I'll go ahead and disable for now.